### DISCUSSION TITLE
always keep wifi on during sleep Android 13

### DISCUSSION THREAD

**Discussion by master3395** [NightscoutFoundation/xDrip#2485] [General]
Hello, i have noticed that with the latest Android 13
You will get annoying messages about enabling "always keep wifi on during sleep".
But this option was removed after Android 12. (Not just Samsung like pointed out on the site below, I use OnePlus)
https://eu.community.samsung.com/t5/galaxy-s21-series/wifi-keep-on-during-sleep-setting-missing-after-latest-update/td-p/5152555

This warning comes from time to time.
I have given xDrip+ all perms that it asks for and more.

```
xDrip+ 
Version: 344deb4-2022.11.16
Code: 202211618
```

![image](https://user-images.githubusercontent.com/4199669/202935185-98aa46e8-7b30-4e6c-b290-4afd5f59780d.png)

**Comment by Navid200** [comment]
Are you getting this notification only when you have xDrip enabled?
The message seems to be a general message and not related to any specific app.

**Reply by master3395** [reply]
Yes, it only happens when i open xDrip+ from time to time.

**Reply by master3395** [reply]
Found the reference here:
https://github.com/NightscoutFoundation/xDrip/search?q=always+keep+wifi (4 pages of results)

Here's an example file:
https://github.com/NightscoutFoundation/xDrip/blob/38fee03be56da6ce9cbe90e08dbe8e4b651c6edc/etc/FixLangs/data.txt

**Reply by Navid200** [reply]
Thanks for reporting this.
I have brought it to the attention of the developer who added this feature a few years ago.

If I understand you correctly, you are happy with the Android settings and the only thing that is bothering you is that this message occasionally pops up.
If that is true, you should be able to select NO, NEVER.

Thanks again

**Reply by master3395** [reply]
> Thanks for reporting this. I have brought it to the attention of the developer who added this feature a few years ago.
> 
> If I understand you correctly, you are happy with the Android settings and the only thing that is bothering you is that this message occasionally pops up. If that is true, you should be able to select NO, NEVER.
> 
> Thanks again

Yeah, it only bothers me when the message comes from time to time, and non of the 3 buttons does anything other than to close the popup or re-direct me to the android settings tab where the settings were before.
But with Android 13 it's removed, so the "never" option doesn't save.
